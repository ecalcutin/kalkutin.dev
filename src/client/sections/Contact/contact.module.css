.contact {
  margin-left: auto;
  margin-right: auto;
  text-align: center;
  max-width: 60rem;
}

.contactButton {
  margin-top: 5rem;
  text-align: center;
}
