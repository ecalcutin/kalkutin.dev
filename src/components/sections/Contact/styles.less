@import (reference) "theme/mixins.less";

.contact {
  max-width: 600px;
  margin: 0 auto 100px;
  text-align: center;

  @media (max-width: 768px) {
    margin: 0 auto 50px;
  }

  .overline {
    display: block;
    margin-bottom: 20px;
    color: @color-accent;
    font-family: @font-mono;
    font-size: @font-size-md;
    font-weight: 400;

    &:before {
      bottom: 0;
      font-size: @font-size-sm;
    }

    &:after {
      display: none;
    }
  }

  .title {
    font-size: clamp(40px, 5vw, 60px);
  }

  .emailLink {
    .button();
    margin-top: 50px;
  }
}
