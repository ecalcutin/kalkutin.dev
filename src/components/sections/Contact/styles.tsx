import styled from "styled-components";

export const StyledContactSection = styled.section`
  max-width: 600px;
  margin: 0 auto 100px;
  text-align: center;

  @media (max-width: 768px) {
    margin: 0 auto 50px;
  }

  .overline {
    display: block;
    margin-bottom: 20px;
    color: var(--green);
    font-family: var(--font-mono);
    font-size: var(--fz-md);
    font-weight: 400;

    &:before {
      bottom: 0;
      font-size: var(--fz-sm);
    }

    &:after {
      display: none;
    }
  }

  .title {
    font-size: clamp(40px, 5vw, 60px);
  }

  .email-link {
    ${({ theme }) => theme.mixins.bigButton};
    margin-top: 50px;
  }
`;
