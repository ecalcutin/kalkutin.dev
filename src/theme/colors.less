@import "variables.less";

@color-primary-dark: darken(@color-primary, 5%);
@color-primary-light: lighten(@color-primary, 5%);
@color-primary-lighten: lighten(@color-primary, 10%);
@color-primary-shadow: darken(@color-primary, 30%);

@color-secondary-dark: darken(@color-secondary, 5%);
@color-secondary-light: lighten(@color-secondary, 5%);
@color-secondary-lighten: lighten(@color-secondary, 20%);

@color-accent-tint: fade(@color-accent, 10%);
