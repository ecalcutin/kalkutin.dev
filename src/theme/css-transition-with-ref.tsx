import React, { cloneElement, ReactElement, useRef } from 'react';
import { CSSTransition } from 'react-transition-group';
import { CSSTransitionProps } from 'react-transition-group/CSSTransition';

type Props = CSSTransitionProps & {
  readonly children: ReactElement<any>;
};

export const CSSTransitionWithRef: React.FC<Props> = props => {
  const { children, ...restProps } = props;

  const ref = useRef<HTMLElement>(null);

  const setRef = (value: HTMLElement | null): void => {
    ref.current = value;
  };

  return (
    // @ts-ignore
    <CSSTransition nodeRef={ref} {...restProps}>
      {cloneElement(children, { ref: setRef })}
    </CSSTransition>
  );
};
