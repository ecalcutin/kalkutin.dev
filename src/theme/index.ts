export { default as theme } from './theme-styles';
export { default as GlobalStyle } from './global';
export { default as mixins } from './mixins';
