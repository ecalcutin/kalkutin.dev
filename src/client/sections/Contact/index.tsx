import React from 'react';

import { Section } from 'client/components/Section';

import styles from './contact.module.css';

export const Contact: React.FC = () => {
  return (
    <Section id='contact' title='Get In Touch' overline="What's next?">
      <div className={styles.contact}>
        <p className={styles.content}>
          Although I&apos;m not currently looking for any new opportunities, my
          inbox is always open. Whether you have a question or just want to say
          hi, I&apos;ll try my best to get back to you!
        </p>
        <p className={styles.contact}>
          <a className='email-link' href={`mailto:evgheni.calcutin@gmail.com`}>
            {/* <Button> Say Hello</Button> */}
          </a>
        </p>
      </div>
    </Section>
  );
};
