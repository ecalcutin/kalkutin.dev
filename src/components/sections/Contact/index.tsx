import React from "react";
import clsx from "clsx";

import styles from "./styles.less";

const Contact = () => {
  return (
    <section id="contact" className={styles.contact}>
      <h2 className={clsx(["numbered-heading", styles.overline])}>
        What's Next?
      </h2>
      <h2 className={styles.title}>Get In Touch</h2>
      <p>
        Although I'm not currently looking for any new opportunities, my inbox
        is always open. Whether you have a question or just want to say hi, I'll
        try my best to get back to you!
      </p>
      <a className={styles.emailLink} href="mailto:evgheni.calcutin@gmail.com">
        Say Hello
      </a>
    </section>
  );
};

export default Contact;
