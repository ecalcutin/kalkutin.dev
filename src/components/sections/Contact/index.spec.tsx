import renderer from "react-test-renderer";
import { ThemeProvider } from "styled-components";

import { theme } from "../../../theme";

import Contact from "./index";

describe("Contact section", () => {
  it("should render correctly", () => {
    const tree = renderer.create(
      <ThemeProvider theme={theme}>
        <Contact />
      </ThemeProvider>
    );
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
