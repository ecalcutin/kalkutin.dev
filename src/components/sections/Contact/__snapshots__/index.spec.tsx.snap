// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact section should render correctly 1`] = `
<section
  className="sc-bdnxRM hylKnQ"
  id="contact"
>
  <h2
    className="numbered-heading overline"
  >
    Whatâ€™s Next?
  </h2>
  <h2
    className="title"
  >
    Get In Touch
  </h2>
  <p>
    Although I'm not currently looking for any new opportunities, my inbox is always open. Whether you have a question or just want to say hi, I'll try my best to get back to you!
  </p>
  <a
    className="email-link"
    href="mailto:evgheni.calcutin@gmail.com"
  >
    Say Hello
  </a>
</section>
`;
